rosserial_python/src/rosserial_python/SerialClient.py

```diff
@@ -334,7 +334,7 @@ def __init__(self, port=None, baud=57600, timeout=5.0, fix_pyserial_for_test=Fal
 
         self.pub_diagnostics = rospy.Publisher('/diagnostics', diagnostic_msgs.msg.DiagnosticArray, queue_size=10)
 
-        rospy.Service('~hard_reset', Empty, self.hard_reset)
+        rospy.Service('~hard_reset_arduino', Empty, self.hard_reset_arduino)
```

Comment:
Pedantic, but isn't this more of a soft reset?

Reply:
I guess it depends on your definition. I would define a soft reset asking the Arduino to reset itself in your sketch, which may or may not be supported. I believe toggling the DTR pin actually pulls the Arduino's reset pin, so it's not effected by the sketch.